dfastmi/io/RiverParameterData.py

```diff
@@ -219,183 +226,175 @@ def collect_values1(
+        val = self.__read_value(key, branch, reach)
+        if val == "" and default is not None:
+            fval = default
+        else:
             try:
-                b_val = self._config[branch][key]
+                vals = tuple(float(x) for x in val.split())
             except:
-                b_val = g_val
-            values_per_branch = []
-            for i in range(self._nreaches[ib]):
-                stri = str(i + 1)
-                try:
-                    val = self._config[branch][key + stri]
-                except:
-                    val = b_val
-                if val == "" and default is not None:
-                    vals = default
-                else:
-                    try:
-                        vals = tuple(float(x) for x in val.split())
-                    except:
-                        vals = ()
-                    if len(vals) != 2:
-                        raise Exception(
-                            'Reading {} for reach {} on {} returns "{}". Expecting {} values.'.format(
-                                key, i+1, self._branches[ib], val, 2
-                            )
-                        )
-                values_per_branch.append((vals[0], vals[1]))
-            all_values.append(values_per_branch)
-        return all_values
+                vals = ()
+            if len(vals) != 2:
+                raise Exception(
+                    'Reading {} for reach {} on branch {} returns "{}". Expecting {} values.'.format(
+                        key, reach.name, branch.name, val, 2
+                    )
+                )
+            fval = (vals[0], vals[1])
+        return fval
     
-    def collect_values4(
-        self,
-        key: str,
-        default: Optional[Tuple[float, float, float, float]] = None,
-    ) -> List[List[Tuple[float, float, float, float]]]:
+    def read_key_tuple_float_float_float_float(
+            self,
+            key: str,
+            branch: Branch,
+            reach: Reach,
+            default: Optional[Tuple[float, float, float, float]] = None,
+        ) ->  Tuple[float, float, float, float]:
         """
-        This routines collects entries of type Tuple[float, float, float, float].
+        This routines collects entries of type float.
 
         Arguments
         ---------
         key : str
             The name of the parameter for which the values are to be retrieved.
-        default : Optional[Tuple[float, float, float, float]]
-            Default tuple if not specified in file.
+        branch : Branch
+            The branch where we want to read the parameter from
+
 
         Raises
         ------
         Exception
-            If the number of values read from the file doesn't match 4.
+            If the number of values read from the file doesn't match 1.
 
         Returns
         -------
-        data : List[List[Tuple[float, float, float, float]]]
+        data : List[List[float]]
```

Comment:
I will stop repeating myself in this file from here on :zipper_mouth_face:

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112